src/components/common/UserProfileEdit/UserProfileEdit.tsx

```diff
@@ -0,0 +1,67 @@
+import React from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+
+import { Tab, TabList, TabPanel, Tabs } from '~shared/Tabs';
+
+import ProfileTemplate from '~frame/ProfileTemplate';
+import LandingPage from '~frame/LandingPage';
+
+import { useAppContext } from '~hooks';
+
+import UserProfileSpinner from '../UserProfile/UserProfileSpinner';
+import Sidebar from './Sidebar';
+import UserMainSettings from './UserMainSettings';
+import UserAdvancedSettings from './UserAdvancedSettings';
+
+const displayName = 'common.UserProfileEdit';
+
+const MSG = defineMessages({
+  headingMain: {
+    id: `${displayName}.headingMain`,
+    defaultMessage: 'User profile',
+  },
+  headingAdvance: {
+    id: `${displayName}.headingAdvance`,
+    defaultMessage: 'Advanced settings',
+  },
+});
+
+const UserProfileEdit = () => {
+  const { user, userLoading } = useAppContext();
+
+  if (userLoading || user === undefined) {
+    return <UserProfileSpinner />;
+  }
+
+  if (user === null) {
```

Comment:
I think this should redirect user (by replacing the URL) rather than just render the landing page

Reply:
I agree, but for the current implementation that wouldn't work. If when loading is fixed I will redirect.